its/ruling/src/test/java/org/sonar/java/it/AutoScanTest.java

```diff
@@ -167,28 +170,49 @@ public void javaCheckTestSources() throws Exception {
     List<IssueDiff> rulesSilenced = newDiffs.stream().filter(IssueDiff::onlyFNs).collect(Collectors.toList());
     LOG.info("{} rules silenced without binaries (only FNs):\n{}", rulesSilenced.size(), IssueDiff.prettyPrint(rulesSilenced));
 
-    // store in a JSON file - serializable
-    Files.writeString(pathFor(TARGET_ACTUAL + DIFF_FILE + ".json"), GSON.toJson(newDiffs));
-    // store in a CSV file - can be easily imported in google sheets
+    // Load known diffs
+    var knownDiffFiles = new ArrayList<Path>();
+    try (var dirStream = Files.newDirectoryStream(pathFor("src/test/resources/autoscan/diffs/"),
+      path -> path.getFileName().toString().startsWith("diff_") && path.toString().endsWith(".json"))) {
+      dirStream.forEach(knownDiffFiles::add);
+    }
+    var knownDiffs = new HashMap<String, IssueDiff>();
+    for (var diffFile : knownDiffFiles) {
+      IssueDiff diff = GSON.fromJson(Files.readString(diffFile), GSON_ISSUE_DIFF_TYPE);
+      knownDiffs.put(diff.ruleKey, diff);
+    }
+
+    // store new unexpected diffs in JSON files - serializable
```

Comment:
For easier debugging, we only output diffs that are different from the expected. There is no value in outputting diffs that are expected that way.